src/main/java/com/textmessenger/service/UserServiceImpl.java

```diff
@@ -234,4 +244,70 @@ public String changePasswordForgot(CredentialsPassword credentialsPassword) {
     }
     return "Sorry. This token in invalid.";
   }
+
+  @Override
+  public void updateUserWithStringsAndFile(String firstName,
+                                           String lastName,
+                                           String address,
+                                           String date,
+                                           MultipartFile file) throws IOException {
+    UserPrincipal userPrincipal = (UserPrincipal) SecurityContextHolder
+            .getContext()
+            .getAuthentication()
+            .getPrincipal();
+    User one = userRepository.getOne(userPrincipal.getId());
+    if (firstName != "undefined") {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "undefined" 3 times. ([squid:S1192](https://sonarcloud.io/organizations/final_project_text_messenger/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=Text_Messenger_Project&pullRequest=194&issues=AWXd3GSh_eEZPtUmH9zH&open=AWXd3GSh_eEZPtUmH9zH)

[ISSUE_KEY]: AWXd3GSh_eEZPtUmH9zH